### PR TITLE
Add icon overlay to shelf inventory slots

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR modifies the vessels shelf and the bookshelf formspecs.
It adds a nice overlay icon to the shelf inventory slots, reminding the player what kind of item is permitted in those slots (since it is not obvious). The style is inspired by the trash slot from the creative inventory.

Note: For the formspec changes to take place of already placed shelves, the players must manually mine and place the shelf again. But since this PR only does a visual change, I hope this is not too serious.

Screenshot of new bookshelf formspec:
Screenshot of new bookshelf formspec

Screenshot of new vessels shelf formspec:
Screenshot of new vessels shelf formspec

### CODE REVIEW

File: mods/vessels/init.lua [hunk 1]
```diff
@@ -13,6 +13,17 @@ local vessels_shelf_formspec =
 	"listring[current_player;main]" ..
 	default.get_hotbar_bg(0, 2.85)
 
+-- Inventory slots overlay
+local vx, vy = 0, 0.3
+for i=1,16 do
```
Comment (sofar) [inline mods/vessels/init.lua]: spaces

File: mods/vessels/init.lua [hunk 2]
```diff
@@ -13,6 +13,17 @@ local vessels_shelf_formspec =
...
+	if i==9 then vx = 0
```
Comment (sofar) [inline mods/vessels/init.lua]: same, spaces

**Comment by sofar** [comment]
two small whitespace comments, :+1: on the idea. does the furnace fuel inventory have a fuel-like backdrop as well? if not then that would be a good one to do as well.

**Comment by Wuzzy2** [comment]
Sofar, what are you talking about? AFAIK there is no coding style guide (at least not in Minetest Game's directory).

File: mods/vessels/init.lua (hunk 2 above)
Comment (sofar) [inline mods/vessels/init.lua]: `i == 9`
`for i = 1, 16 do`

**Comment by sofar** [comment]
My edit shows as a commit on your branch? I'm not sure I like that :/

Anyway, a squash on merge fixes and this is :+1:

**Comment by Wuzzy2** [comment]
sofar: Yes, the change is on my branch. I am OK with this and I am free to disallow this, if neccessary. Also, thanks. :-)

**Comment by rubenwardy** [comment]
I like the concept but have not looked at the code yet.

**Comment by sfan5** [comment]
:+1: for the concept

**Comment by Napiophelios** [comment]
I think this is kind of tacky looking. Especially in the vessels formspec, 
the mock images confuse the transparent inventory images if its not a potion bottle.
Maybe it would be better to message the chat when players try to add a disallowed item.

**Comment by Wuzzy2** [comment]
> Maybe it would be better to message the chat when players try to add a disallowed item.

No, no, no! Please let's not spam the chat with messages like this. IMO the chat should only be reserved for … well, chat. And a few *important* events. But nothing else.
I would rather have this PR nuked than adding spammy chat messages.

**Comment by sofar** [comment]
You can maybe fade the images a bit more, to make them less obtrusive, but I wouldn't worry too much about that. Texture packs can also override and largely disable this.

**Comment by tobyplowy** [comment]
@Wuzzy2 Can the background image be disabled whin a item is placed in the slot? (for example The cup Looks a bit weird with the gray background image of the bottle)

**Comment by Wuzzy2** [comment]
@sofar: I chose the same color as the trash slot. I won't change the icons without further opinions.
@tobyplowy: No.

**Comment by sofar** [comment]
The same brightness as the trash slot is fine with me.

**Comment by tobyplowy** [comment]
@Wuzzy2 why don't we Add a proper gui interface instead of the boring dark grey box we have now (images as gui)

Also why No?

**Comment by Wuzzy2** [comment]
@tobyplowy: If you have suggestions for the redesign of the GUI, open a new issue. A complete GUI rework is obviously out of scope of this PR.
“No” because it is not technically possible with the current formspec API.

**Comment by tobyplowy** [comment]
@Wuzzy2 I already opened An isshu on that a while ago and paramat said it was a mod thing :/

**Comment by sofar** [comment]
@tobyplowy the inventory already is "proper".

**Comment by paramat** [comment]
This looks messy for the vessels shelf and i would rather you don't bother with the vessels shelf since we're thinking of removing vessels mod from MTGame.
I'm neutral about the usage for bookshelf.

**Comment by C1ffisme** [comment]
@paramat I wouldn't remove all vessels. Removing the steel bottle and the glass cup is fine, since mods (AFAIK) do not use them, and the normal bottle is there for if they depend on a handheld smaller liquid storage. Vessels shelves can stay, if we ever get potions/drinkables.

I like the overlays. While it is obvious to anyone who tries to place items in a shelf that don't belong, that it only accepts books/bottles (If the name wasn't enough :stuck_out_tongue:), it's like it gives more feedback. It's almost as if the slots were carved for the items, so it makes since why only they fit in there. (Except for multiple bottles, oh well. You can't win them all.)

**Comment by Wuzzy2** [comment]
Please stay on topic. I did not intend to remove anything.

**Comment by paramat** [comment]
That's my fault, i deeted a comment of mine.

**Comment by C1ffisme** [comment]
Went ahead and deleted mine too. That's out of scope for this pull request.

**Comment by everamzah** [comment]
@Napiophelios is right, the one background image for vessel shelf is different than the two others, causing a visible overlap.

image

**Comment by paramat** [comment]
The bookshelf backgrounds seem good to me, i don't care about vessels much so not opposed to that.

**Comment by everamzah** [comment]
>I don't care about vessels much so not opposed to that.

@paramat The vessel overlay image is quite unfitting.

**Comment by tobyplowy** [comment]
@everamzah That's why the overlay should be disabled if something is in the item slot

**Comment by sofar** [comment]
Counting @sfan5's approval as a full approval (since it's just textures) and mine, this has 2 approvals already.

**Comment by rubenwardy** [comment]
@sfan5 approved the concept not the PR

File: mods/default/nodes.lua [hunk 3]
```diff
@@ -1807,6 +1807,18 @@ local bookshelf_formspec =
...
 
+-- Inventory slots overlay
+local bx, by = 0, 0.3
+for i = 1, 16 do
+	if i == 9 then
+		bx = 0
+		by = by + 1
+	end
+	bookshelf_formspec = bookshelf_formspec ..
+	"image["..bx..","..by..";1,1;default_book_slot.png]"
```
Comment (rubenwardy) [inline mods/default/nodes.lua]: one more indent here please

File: mods/vessels/init.lua [hunk 4]
```diff
@@ -13,6 +13,17 @@ local vessels_shelf_formspec =
...
+		vy = vy + 1
+	end
+	vessels_shelf_formspec = vessels_shelf_formspec ..
+	"image["..vx..","..vy..";1,1;vessels_glass_bottle_slot.png]"
```
Comment (rubenwardy) [inline mods/vessels/init.lua]: here too

**Comment by Wuzzy2** [comment]
OK, code style applied. By the way, is the coding style written down somewhere?

**Comment by rubenwardy** [comment]
You missed the ones sofar pointed out, but don't worry, that could be changed on merge

**Comment by rubenwardy** [comment]
http://dev.minetest.net/Lua_code_style_guidelines

**Comment by sofar** [comment]
0acd9a2

**Pull request closed without merging**